src/utils/addresses.ts

```diff
@@ -0,0 +1,35 @@
+import { getAddress } from '@ethersproject/address'
+
+// returns the checksummed address if the address is valid, otherwise returns false
+export function isAddress(value: any): string | false {
+  try {
+    // Alphabetical letters must be made lowercase for getAddress to work.
+    // See documentation here: https://docs.ethers.io/v5/api/utils/address/
+    return getAddress(value.toLowerCase())
+  } catch {
+    return false
+  }
+}
+
+/**
+ * Shortens an Ethereum address by N characters (search keywords: .slice .truncate)
+ * @param address blockchain address
+ * @param charsStart amount of character to shorten (from both ends / in the beginning)
+ * @param charsEnd amount of characters to shorten in the end
+ * @returns formatted string
+ */
+export function shortenAddress(address: string, charsStart = 4, charsEnd?: number): string {
```

Comment:
lol semantically this one feels more like `shortenAddressIfValid` than the other one - since the other one errors, it assumes the address has to be valid 🤔

Reply:
hm my intuition says the `if` sort of indicates it fails, i was thinking `UnlessInvalid` but realize thats the same thing...

sometimes i've seen `shortenAddressSafe` used for the one that always returns, but that's a bit icky too.

`shortenAddressStrict` perhaps for the one that throws?

Reply:
went with Strict for now i kind of like it